options/options.c

```diff
@@ -544,44 +557,56 @@ static const m_option_t mp_opts[] = {
         {"normal",      NORMAL_PRIORITY_CLASS},
         {"belownormal", BELOW_NORMAL_PRIORITY_CLASS},
         {"idle",        IDLE_PRIORITY_CLASS}),
-        .flags = UPDATE_PRIORITY},
-#endif
+        .flags = UPDATE_PRIORITY,
+        .not_available = !HAVE_WIN32_DESKTOP},
+
+    // Windows Medial Controls
     {"media-controls", OPT_CHOICE(media_controls,
-        {"no", 0}, {"player", 1}, {"yes", 2})},
+        {"no", 0}, {"player", 1}, {"yes", 2}),
+        .not_available = !HAVE_WIN32_SMTC},
+
     {"config", OPT_BOOL(load_config), .flags = CONF_PRE_PARSE},
     {"config-dir", OPT_STRING(force_configdir),
         .flags = CONF_NOCFG | CONF_PRE_PARSE | M_OPT_FILE},
     {"reset-on-next-file", OPT_STRINGLIST(reset_options)},
 
-#if HAVE_LUA || HAVE_JAVASCRIPT || HAVE_CPLUGINS
-    {"scripts", OPT_PATHLIST(script_files), .flags = M_OPT_FILE},
-    {"script", OPT_CLI_ALIAS("scripts-append")},
-    {"script-opts", OPT_KEYVALUELIST(script_opts)},
-    {"load-scripts", OPT_BOOL(auto_load_scripts)},
-#endif
-#if HAVE_JAVASCRIPT
-    {"js-memory-report", OPT_BOOL(js_memory_report)},
-#endif
-#if HAVE_LUA
-    {"osc", OPT_BOOL(lua_load_osc), .flags = UPDATE_BUILTIN_SCRIPTS},
-    {"ytdl", OPT_BOOL(lua_load_ytdl), .flags = UPDATE_BUILTIN_SCRIPTS},
-    {"ytdl-format", OPT_STRING(lua_ytdl_format)},
-    {"ytdl-raw-options", OPT_KEYVALUELIST(lua_ytdl_raw_options)},
-    {"load-stats-overlay", OPT_BOOL(lua_load_stats),
-        .flags = UPDATE_BUILTIN_SCRIPTS},
-    {"load-osd-console", OPT_BOOL(lua_load_console),
-        .flags = UPDATE_BUILTIN_SCRIPTS},
+    // Scripting
+    {"scripts", OPT_PATHLIST(script_files), .flags = M_OPT_FILE,
+        .not_available = !HAVE_LUA && !HAVE_JAVASCRIPT && !HAVE_CPLUGINS},
+    {"script", OPT_CLI_ALIAS("scripts-append"),
+        .not_available = !HAVE_LUA && !HAVE_JAVASCRIPT && !HAVE_CPLUGINS},
+    {"script-opts", OPT_KEYVALUELIST(script_opts),
+        .not_available = !HAVE_LUA && !HAVE_JAVASCRIPT && !HAVE_CPLUGINS},
+    {"load-scripts", OPT_BOOL(auto_load_scripts),
+        .not_available = !HAVE_LUA && !HAVE_JAVASCRIPT && !HAVE_CPLUGINS},
+
+    // Javascript
+    {"js-memory-report", OPT_BOOL(js_memory_report), .not_available = !HAVE_JAVASCRIPT},
+
+    // Lua
+    {"osc", OPT_BOOL(lua_load_osc), .flags = UPDATE_BUILTIN_SCRIPTS,
```

Comment:
Now that it will not be hard error we can remove this from test
```
diff --git a/test/meson.build b/test/meson.build
index 44983f18d0..1b7bf11ff6 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -119,7 +119,7 @@ if get_option('libmpv')
     exe = executable('libmpv-test', 'libmpv_test.c',
                      include_directories: incdir, link_with: libmpv)
     file = join_paths(source_root, 'etc', 'mpv-icon-8bit-16x16.png')
-    test('libmpv', exe, args: file, timeout: 60, should_fail: not features['lua'])
+    test('libmpv', exe, args: file, timeout: 60)
 
     exe = executable('libmpv-encode', 'libmpv_encode.c',
                      include_directories: incdir, link_with: libmpv)

```

Reply:
No that test still fails which I'd expect because loading the osc won't work.